sklearnex/preview/linear_model/ridge.py

```diff
@@ -0,0 +1,343 @@
+        self._onedal_estimator.intercept_ = self.intercept_
+
+    def _onedal_fit(self, X, y, sample_weight, queue=None):
+        assert sample_weight is None
+
+        check_params = {
+            "X": X,
+            "y": y,
+            "dtype": [np.float64, np.float32],
+            "accept_sparse": ["csr", "csc", "coo"],
+            "y_numeric": True,
+            "multi_output": True,
+            "force_all_finite": False,
+        }
+        if sklearn_check_version("1.2"):
+            X, y = self._validate_data(**check_params)
+        else:
+            X, y = check_X_y(**check_params)
+
+        if sklearn_check_version("1.0") and not sklearn_check_version("1.2"):
+            self._normalize = _deprecate_normalize(
+                self.normalize,
+                default=False,
+                estimator_name=self.__class__.__name__,
+            )
+
+        self._initialize_onedal_estimator()
+        try:
+            # Falling back to daal4py if the device is CPU since
+            # onedal does not support non-scalars for alpha, thus
+            # should only be used for GPU to not limit the functionality
+            cpu_device = queue is None or queue.sycl_device.is_cpu
+            if cpu_device:
+                self._daal_fit(X, y)
+            else:
+                self._onedal_estimator.fit(X, y, queue=queue)
+            self._save_attributes()
+
+        except RuntimeError:
+            logging.getLogger("sklearnex").info(
+                f"{self.__class__.__name__}.fit "
+                + get_patch_message("sklearn_after_onedal")
+            )
+
+            del self._onedal_estimator
+            super().fit(X, y)
+
+    def _onedal_predict(self, X, queue=None):
+        if sklearn_check_version("1.0"):
+            self._check_feature_names(X, reset=False)
+
+        X = self._validate_data(X, accept_sparse=False, reset=False)
+        if not hasattr(self, "_onedal_estimator"):
+            self._initialize_onedal_estimator()
+            self._onedal_estimator.coef_ = self.coef_
+            self._onedal_estimator.intercept_ = self.intercept_
+
+        res = self._onedal_estimator.predict(X, queue=queue)
+        return res
+
+    def get_coef_(self):
+        return self.coef_
+
+    def set_coef_(self, value):
+        self.__dict__["coef_"] = value
+        if hasattr(self, "_onedal_estimator"):
+            self._onedal_estimator.coef_ = value
+            del self._onedal_estimator._onedal_model
+
+    def get_intercept_(self):
+        return self.intercept_
+
+    def set_intercept_(self, value):
+        self.__dict__["intercept_"] = value
+        if hasattr(self, "_onedal_estimator"):
+            self._onedal_estimator.intercept_ = value
+            del self._onedal_estimator._onedal_model
+
+    def _save_attributes(self):
+        self.coef_ = property(self.get_coef_, self.set_coef_)
+        self.intercept_ = property(self.get_intercept_, self.set_intercept_)
```

Comment:
Use the trick elsewhere to solve this (raising a specific attribute error will false out hasattr).